core/modules/file/file.field.inc

```diff
@@ -282,7 +282,7 @@ function file_field_update($entity_type, $entity, $field, $instance, $langcode,
   foreach ($items as $item) {
     if (!in_array($item['fid'], $original_fids)) {
       if ($file = file_load($item['fid'])) {
-        file_usage_add($file, 'file', $entity_type, $id);
+        file_usage_add($file, 'file', $entity_type, $item['fid']);
```

Comment:
I suspect, this is not the right approach.

$id gets set in line 247 and should be - according to the comment block of function file_usage_add():

> The unique, numeric ID of the object containing the referenced file.

Not the file fid.